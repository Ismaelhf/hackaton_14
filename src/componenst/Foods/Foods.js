import React from 'react';
import {
  StyleSheet,
  View,
  Image,
  TouchableHighlight,
  TouchableOpacity,
  Text,
  ScrollView,
} from 'react-native';

const Foods = ({data, navigation}) => {
  return (
    <>
      {data.map((item) => (
        <View style={styles.listadoItem} key={item.id}>
          <TouchableOpacity
            onPress={() => navigation.navigate('Foods', {item: {item}})}>
            <Image
              style={styles.mejores}
              source={{
                uri: item.image,
              }}
            />
          </TouchableOpacity>
        </View>
      ))}
    </>
  );
};

const styles = StyleSheet.create({
  listado: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
  },
  listadoItem: {
    flexBasis: '49%',
  },
  mejores: {
    width: '100%',
    height: 200,
    marginVertical: 5,
  },
});
export default Foods;
