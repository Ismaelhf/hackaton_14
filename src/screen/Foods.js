import React from 'react';
import {StyleSheet, View, Image, Text} from 'react-native';

const Foods = ({route}) => {
  const item = route.params.item.item;
  // console.warn(item.title);
  return (
    <View style={styles.item}>
      <View>
        <Image
          style={styles.ciudad}
          source={{
            uri: item.image,
          }}
        />
      </View>

      <View style={styles.text}>
        <Text style={styles.title}>{item.title}</Text>

        <Text style={styles.desp}>{item.description}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  item: {
    flex: 1,
  },
  ciudad: {
    width: '100%',
    height: 300,
  },

  text: {
    marginHorizontal: 20,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 20,
    textAlign: 'center',
  },
  desp: {
    textAlign: 'justify',
  },
});

export default Foods;
