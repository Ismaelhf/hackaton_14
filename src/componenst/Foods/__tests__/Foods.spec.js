// import { shallow, mount, render } from 'enzyme';
import React from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import {shallow} from 'enzyme';
import Foods from '../Foods';
import {foodsData} from '../../../data/foodsData';

const initialProps = {
  data: foodsData,
};

// console.log('test');
const wrapper = shallow(<Foods {...initialProps} />);

describe('Foods component', () => {
  //   beforeEach(() => {
  //     wrapper = shallow(<Foods {...initialProps} />);
  //   });

  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should Foods element in component', () => {
    expect(wrapper.find(View).length).toBe(4);
    expect(wrapper.find(TouchableOpacity).length).toBe(4);
    expect(wrapper.find(Image).length).toBe(4);
  });
  it('should match props', () => {
    expect(wrapper.props.data).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
