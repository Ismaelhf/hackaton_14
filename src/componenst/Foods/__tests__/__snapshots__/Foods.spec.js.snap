// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foods component should match snapshot 1`] = `
<Fragment>
  <View
    key="1"
    style={
      Object {
        "flexBasis": "49%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "uri": "https://cdn.viajala.com/img/blog/5-platos-tipicos-peru/ceviche-peruano.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </ForwardRef>
  </View>
  <View
    key="2"
    style={
      Object {
        "flexBasis": "49%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "uri": "https://img-s-msn-com.akamaized.net/tenant/amp/entityid/BB15poFh.img?h=1080&w=1920&m=6&q=60&o=t&l=f",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </ForwardRef>
  </View>
  <View
    key="3"
    style={
      Object {
        "flexBasis": "49%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "uri": "https://i.pinimg.com/originals/9d/f7/59/9df759ad0e787a6639f75ec22a113c92.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </ForwardRef>
  </View>
  <View
    key="4"
    style={
      Object {
        "flexBasis": "49%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSWyXQ7fOeIoH_jUQBsVpv6pmzws6ynV6WDEg&usqp=CAU",
          }
        }
        style={
          Object {
            "height": 200,
            "marginVertical": 5,
            "width": "100%",
          }
        }
      />
    </ForwardRef>
  </View>
</Fragment>
`;
